app/web/src/components/ModelingView/FloatingConnectionMenu.vue

```diff
@@ -528,157 +576,171 @@ watch(componentsStore.allComponentsById, debouncedUpdate, { immediate: true });
 const socketlessConnectionCandidatesA = ref<ConnectionCandidateProp[]>([]);
 const socketlessConnectionCandidatesB = ref<ConnectionCandidateProp[]>([]);
 
-const fullASideList = computed(() => {
-  return [...listAItems.value, ...socketlessConnectionCandidatesA.value];
-});
+const fullASideList = computed(() => [
+  ...listAItems.value,
+  ...socketlessConnectionCandidatesA.value,
+]);
 
 const fullBSideList = computed(() => {
+  if (connectionData.A.socketId) {
+    return [...listBItems.value];
+  } else if (connectionData.A.attributePath) {
+    return [...socketlessConnectionCandidatesB.value];
+  }
+
   return [...listBItems.value, ...socketlessConnectionCandidatesB.value];
 });
 
 // Prop candidates
 const schemaVariantsById = reactive({} as Record<string, PropertyEditorSchema>);
 const storedInitialComponent = ref();
 const loadingPropConnections = ref(false);
-watch([storedInitialComponent, searchStrings], async () => {
-  if (!featureFlagsStore.PROPS_TO_PROPS_CONNECTIONS) {
-    return;
-  }
+const loadedPropConnections = ref(false);
+watch(
+  [storedInitialComponent, searchStrings, connectionData.aDirection],
+  async () => {
```

Comment:
One interesting thing about watch (not a thing that needs a-changin'): the things you watch will have their current values passed in as parameters to the lambda.